src/main/src/test/java/org/geoserver/logging/LoggingStartupContextListenerTest.java

```diff
@@ -127,6 +128,26 @@ public void testLogLocationFromServletContext() throws Exception {
         }
     }
 
+    @Test
+    public void testLogLocationFromEmptyContext() throws Exception {
+        File tmp = File.createTempFile("log", "tmp", new File("target"));
+        tmp.delete();
+        tmp.mkdirs();
+
+        File logs = new File(tmp, "logs");
+        assertTrue(logs.mkdirs());
+
+        MockServletContext context = new MockServletContext();
+        context.setInitParameter("GEOSERVER_DATA_DIR", tmp.getPath());
+        context.setInitParameter(
+                "GEOSERVER_LOG_LOCATION", new File(tmp, "foo.log").getAbsolutePath());
+
+        ServletContextListener listener = new LoggingStartupContextListener();
+
+        listener.contextInitialized(new ServletContextEvent(context));
```

Comment:
The test case does not seem to listen to the logs to see if an logging event is generated. I made an test appender for writing these kind of test cases; but this is a bit of a special case since there is not yet an existing logging setup to add a test appender to! 

So this test case is good and we should keep it; but it does not check that logging messages for the log file being missing are indeed suppressed.

Reply:
@dieterstueken can you add the missing tests?

Reply:
will have a look this evening ...

Reply:
My initial intention was just to avoid the NPE.
(sorry for the different issues flipping back and forth, 
something with the badge processing i did not get right)

Without the fix the NPE is raised and after the fix it vanished. So, the test works fine for me.

That the logging generation must be suppressed if the config file is missing, is an additional requirement i was not aware of. But I may simply add a test if the log directory really remains empty after all.

I also observed that the test does not finally remove the directories created (but this is true for all other tests, too).

Reply:
Not removing the directories created is an intentional decision, allows the developer to inspect the data dir contents used by the tests. Very important when pulling hairs debugging a complex situation. The directories are placed in ``/target``, which is meant to contain removable information anyways (so ``mvn clean`` will get rid of them).

Reply:
Here is the code snippet - however TestAppender was not collecting events when I tried:

```
try (TestAppender appender = TestAppender.createAppender("quite", null)) {
            appender.startRecording("org.geoserver.logging");
            
            appender.trigger("Could not reconfigure LOG4J loggers");
            
            LoggingStartupContextListener listener = new LoggingStartupContextListener();
            listener.contextInitialized(new ServletContextEvent(context));
            listener.contextDestroyed(new ServletContextEvent(context));

            appender.stopRecording("org.geoserver.logging");
        }
```

The code snippet is correct though, and when TestAppender functionality is fixed it will be correct.